src/language-server/project/base.ts

```diff
@@ -120,29 +121,32 @@ export abstract class GraphQLProject {
   abstract includesFile(uri: DocumentUri): boolean;
 
   abstract onDidChangeWatchedFiles: ConnectionHandler["onDidChangeWatchedFiles"];
-  abstract onDidOpen?: (event: TextDocumentChangeEvent<TextDocument>) => void;
-  abstract onDidClose?: (event: TextDocumentChangeEvent<TextDocument>) => void;
```

Comment:
Removing the `abstract` here for the optional methods means we can remove the weird `undefined` definitions where they are not implemented.